pkg/test/main_entry.go

```diff
@@ -38,21 +49,63 @@ func MainEntry(m *testing.M) {
 	globalManPath := flag.String(GlobalManPathFlag, "", "path to operator manifest")
 	namespacedManPath := flag.String(NamespacedManPathFlag, "", "path to rbac manifest")
 	singleNamespace = flag.Bool(SingleNamespaceFlag, false, "enable single namespace mode")
+	localOperator := flag.Bool(LocalOperatorFlag, false, "enable if operator is running locally (not in cluster)")
 	flag.Parse()
 	// go test always runs from the test directory; change to project root
 	err := os.Chdir(*projRoot)
 	if err != nil {
 		log.Fatalf("failed to change directory to project root: %v", err)
 	}
-	if err := setup(kubeconfigPath, namespacedManPath); err != nil {
+	if err := setup(kubeconfigPath, namespacedManPath, *localOperator); err != nil {
 		log.Fatalf("failed to set up framework: %v", err)
 	}
+	// setup local operator command, but don't start it yet
+	var localCmd *exec.Cmd
+	var localCmdOutBuf, localCmdErrBuf bytes.Buffer
+	if *localOperator {
+		// taken from commands/operator-sdk/cmd/up/local.go
+		runArgs := append([]string{"run"}, []string{filepath.Join(scaffold.ManagerDir, scaffold.CmdFile)}...)
+		localCmd = exec.Command("go", runArgs...)
+		localCmd.Stdout = &localCmdOutBuf
+		localCmd.Stderr = &localCmdErrBuf
+		c := make(chan os.Signal)
+		signal.Notify(c, os.Interrupt, syscall.SIGTERM)
+		go func() {
+			<-c
+			err := localCmd.Process.Signal(os.Interrupt)
+			if err != nil {
+				log.Fatalf("failed to terminate the operator: (%v)", err)
+			}
+			os.Exit(0)
+		}()
+		if *kubeconfigPath != "" {
```

Comment:
Moving this check before the localOperator check, helps us to fail early.

Reply:
This isn't a check that results in a failure, it just configures some environment variables. If the user provided their own kubeconfig path we set the kubeconfig env var to that, otherwise we use `clientcmd` to get the default location for the kubeconfig.